tests/linearGaussian_snle_test.py

```diff
@@ -502,3 +502,108 @@ def test_api_nle_sampling_methods(
+
+    if prior_str == "gaussian":
+        prior_mean = zeros(num_dim)
+        prior_cov = eye(num_dim)
+        prior = MultivariateNormal(loc=prior_mean, covariance_matrix=prior_cov)
+    else:
+        prior = BoxUniform(-2.0 * ones(num_dim), 2.0 * ones(num_dim))
+
+    def simulator(theta):
+        return linear_gaussian(theta, likelihood_shift, likelihood_cov)
+
+    # Use likelihood_nn with z_score_theta="transform_to_unconstrained"
+    density_estimator = likelihood_nn(
+        model_str,
+        hidden_features=60,
+        num_transforms=3,
+        z_score_theta="transform_to_unconstrained",
+        x_dist=prior,
+    )
+    inference = NLE(density_estimator=density_estimator)
+
+    theta = prior.sample((num_simulations,))
+    x = simulator(theta)
+
+    likelihood_estimator = inference.append_simulations(theta, x).train()
+
+    # Test inference amortized over trials.
+    for num_trials in trials_to_test:
+        x_o = zeros((num_trials, num_dim))
+        if prior_str == "gaussian":
+            gt_posterior = true_posterior_linear_gaussian_mvn_prior(
+                x_o, likelihood_shift, likelihood_cov, prior_mean, prior_cov
+            )
+            target_samples = gt_posterior.sample((num_samples,))
+        elif prior_str == "uniform":
+            target_samples = samples_true_posterior_linear_gaussian_uniform_prior(
+                x_o,
+                likelihood_shift,
+                likelihood_cov,
+                prior=prior,
+                num_samples=num_samples,
+            )
+        else:
+            raise ValueError(f"Wrong prior_str: '{prior_str}'.")
+
+        potential_fn, theta_transform = likelihood_estimator_based_potential(
+            prior=prior, likelihood_estimator=likelihood_estimator, x_o=x_o
+        )
+        posterior = MCMCPosterior(
+            proposal=prior,
+            potential_fn=potential_fn,
+            theta_transform=theta_transform,
+            method="slice_np_vectorized",
+            **mcmc_params_accurate,
+        )
+
+        samples = posterior.sample(sample_shape=(num_samples,))
+
+        # Check performance based on c2st accuracy.
+        check_c2st(
+            samples,
+            target_samples,
+            alg=f"nle_a-{prior_str}-prior-{model_str}-{num_trials}-trials",
+        )
+
+        map_ = posterior.map(
+            num_init_samples=1_000,
+            init_method="proposal",
+            show_progress_bars=False,
+        )
+
+        if prior_str == "uniform":
+            # Check whether the returned probability outside of the support is zero.
+            posterior_prob = get_prob_outside_uniform_prior(posterior, prior, num_dim)
+            assert posterior_prob == 0.0, (
+                "The posterior probability outside of the prior support is not zero"
+            )
+
+            assert ((map_ - ones(num_dim)) ** 2).sum() < 0.5
+        else:
+            assert ((map_ - gt_posterior.mean) ** 2).sum() < 0.5
```

Comment:
I think we can drop this from the test here as well because it is tested in the standard c2st test for nle. Here, we just check whether logit space training gives good enough c2st.